tensorflow/src/TensorFlow/Session.hs

```diff
@@ -199,8 +217,8 @@ runWithFeeds_ feeds t = do
     ns <- build $ getNodes t
     runFetchWithFeeds feeds ns (pure ())
 
-fixFeeds :: [Feed] -> [(ByteString, FFI.TensorData)]
-fixFeeds = map $ \(Feed o d) -> (encodeUtf8 $ encodeOutput o, d)
+fixFeeds :: [Feed] -> [(String, FFI.TensorData)]
```

Comment:
is there a compelling reason to switch from `ByteString` to `String` throughout this PR? at least in the FFI code it is suspicious to me because were going to have to immediately pack the strings back into a contiguous array format for the C APIs to understand it

Reply:
I wanted to use the `%` modifier in a [c2hs function hook](https://github.com/haskell/c2hs/wiki/Implementation-of-Haskell-Binding-Modules#function-hooks) because binding to functions with (non-pointer) struct parameters is not otherwise possible without a wrapper function. Switching to `String` seemed idiomatic because it appears in the list with default mashallers and `ByteString` does not, but I now see that it may be better to revert back to `ByteString` and specify the marshaller in the function hook.

Reply:
i see. i've forgotten most of how c2hs works. i remember avoiding `{# fun .. #}` for some reason, maybe just out of paranoia because it was hard to tell exactly what it was doing